Client/Helpers/FxALoginHelper.swift

```diff
@@ -0,0 +1,191 @@
+        sharedInstance = FxALoginHelper(application: application, profile: profile)
+        return sharedInstance!
+    }
+
+    init(application: UIApplication?, profile: Profile?) {
+        let configuration = DeveloperPushConfiguration()
+        let client = PushClient(endpointURL: configuration.endpointURL)
+
+        self.application = application
+        self.pushClient = client
+        self.profile = profile
+    }
+
+    func userDidLogin(data: JSON) -> Success {
+        if data["keyFetchToken"].asString == nil || data["unwrapBKey"].asString == nil {
+            // The /settings endpoint sends a partial "login"; ignore it entirely.
+            NSLog("Ignoring didSignIn with keyFetchToken or unwrapBKey missing.")
+            self.loginDidFail()
+            return deferred
+        }
+
+        // TODO: Error handling.
+        guard let profile = profile,
+            let account = FirefoxAccount.fromConfigurationAndJSON(profile.accountConfiguration, data: data) else {
+            self.loginDidFail()
+            return deferred
+        }
+
+        self.account = account
+        requestUserNotifications()
+        return deferred
+    }
+
+    func applicationDidLoadProfile() -> Success {
+        guard AppConstants.MOZ_FXA_PUSH else {
+            return finishNormally()
+        }
+
+        guard let account = profile?.getAccount() else {
+            // There's no account, no further action.
+            return finishNormally()
+        }
+
+        if let _ = account.pushRegistration {
+            // We have an account, and it's already registered for push notifications.
+            return finishNormally()
+        }
+
+        // Now: we have an account that does not have push notifications set up.
+        // however, we need to deal with cases of asking for permissions too frequently.
+        let asked = profile?.prefs.boolForKey(applicationDidRequestUserNotificationPermissionPrefKey) ?? true
+        let permitted = application!.currentUserNotificationSettings()!.types != .None
+
+        // If we've never asked(*), then we should probably ask.
+        // If we've asked already, then we should not ask again.
+        // TODO: add UI to tell the user to go flip the Setting app.
+        // (*) if we asked in a prior release, and the user was ok with it, then there is no harm asking again.
+        // If the user denied permission, or flipped permissions in the Settings app, then 
+        // we'll bug them once, but this is probably unavoidable.
+        if asked && !permitted {
+            return finishNormally()
+        }
+
+        // By the time we reach here, we haven't registered for APNS
+        // Either we've never asked the user, or the user declined, then re-enabled
+        // the notification in the Settings app.
+
+        self.account = account
+        requestUserNotifications()
+        return deferred
+    }
+
+    private func requestUserNotifications() {
```

Comment:
Nit: Super small nit but there might be an opportunity to clean this setup code up but moving it to a private extension on `UIUserNotificationSettings` since it's auxiliary to the logic flow of this helper.